recipe/meta.yaml

```diff
@@ -27,42 +27,100 @@ source:
     - nvcc.profile.patch.win  # [win]
 
 build:
-  number: 1
+  number: 2
   skip: true  # [osx or win]
 
-requirements:
-  build:
-    - {{ compiler("c") }}
-    - {{ compiler("cxx") }}
-    - arm-variant * {{ arm_variant_type }}  # [aarch64]
-    - sysroot_{{ target_platform }} 2.17    # [linux]
-  host:
-    - cuda-version {{ cuda_version }}
-  run:
-    - {{ pin_compatible("cuda-version", max_pin="x.x") }}
-    - cuda-cudart-dev
-  run_constrained:
-    - gcc_impl_{{ target_platform }} >=6,<13  # [linux]
-    - arm-variant * {{ arm_variant_type }}  # [aarch64]
-    - cmake >=3.20
+outputs:
+  - name: cuda-nvcc
+    files:
+      - bin                                               # [linux]
+      - targets/{{ target_name }}/bin                     # [linux]
+      - targets/{{ target_name }}/nvvm                    # [linux]
+      - lib/libnvptxcompiler_static.a                     # [linux]
+      - nvvm                                              # [linux]
+      - etc/conda/activate.d/~cuda-nvcc*_activate.sh      # [linux]
+      - etc/conda/deactivate.d/~cuda-nvcc*_deactivate.sh  # [linux]
```

Comment:
Do we want activation scripts for both packages or just one (like `cuda-nvcc_{{ target_platform }}`)?

Reply:
Both packages meaning...?

Reply:
I think we only want them in the host package, unless we need to have compilers for the target architecture as well?

Reply:
I assume this is referring to the `cuda-nvcc` vs `cuda-nvcc_{{ target_platform }}` packages. Unless we need different behaviors by target why would we want it in the latter rather than the former?

Reply:
Yeah exactly. The `*` makes it seem like we are producing and shipping activation scripts for both. Though think we only need the activation script in one place

Personally would suggest `cuda-nvcc_{{ target_platform }}` since that output would be used by things like `{{ compiler("cuda") }}` (as we've seen in our testing recently). Also `cuda-nvcc` would benefit from that indirectly since it requires `cuda-nvcc_{{ target_platform }}`

Would also suggest we replace `PKG_NAME` here with `cuda-nvcc` (what it is now) or something similar. Think `PKG_NAME` may be ambiguous when we move to split packages (as done in this PR)

https://github.com/conda-forge/cuda-nvcc-feedstock/blob/eb3b0973126c6d8c30cff6237f262cc8a8e7a9b3/recipe/build.sh#L47

Reply:
My question is about cross compilation. If `cuda-nvcc_{{ target_platform }}` provides the activation script there would be `cbin` flags added for both host and target `CXX` as I would expect 2 `cuda-nvcc_{{ target_platform }}`s to be in the environment. I would think you would only want host (ie `cuda-nvcc`), unless I am missing something.

Agreed we can change `PKG_NAME`, it resolves to `cuda-nvcc-split`, which is the point of the `*` (not that it generates for both outputs).

Reply:
This is a good point.

With the GCC compiler package, we have [`gcc_{{ ctng_target_platform }}`]( https://github.com/conda-forge/ctng-compiler-activation-feedstock/blob/85ab9f32b2868eddda3280956f5abcc8b8c7a021/recipe/meta.yaml#L21 ), which does activation, and [`gcc_impl_{{ target_platform }}`]( https://github.com/conda-forge/ctng-compilers-feedstock/blob/8118bec940e44fe5e58eddc45a2650fbdf4609e2/recipe/meta.yaml#L115 ), which does not. Maybe we could adopt a similar approach here? Namely have a `cuda-nvcc_impl_{{ target_platform }}` that does not do activation, but is a dependency of `cuda-nvcc_{{ target_platform }}`, which does?

Ah ok. Thanks for clarifying what was going on with `PKG_NAME`. Yeah naming activation scripts this way is just a convention and has the added benefit of avoiding clobbering one another. Think if we went with `~cuda-nvcc_${CHANGE}.sh` there shouldn't be any risk of clobbering (plus this lines up with the file name pre-split)